layers/dns.go

```diff
@@ -464,16 +465,32 @@ func (rr *DNSResourceRecord) String() string {
 	return "..."
 }
 
+func decodeCharacterStrings(data []byte) ([][]byte, error) {
+	var strings [][]byte
```

Comment:
RFC says that TXT has at least one string, but currently gopacket would not raise an error for a malformed packet with an empty TXT: in this case rr.TXT and rr.TXTs would be empty rather than nil slices, but most applications will not see a difference and I do not think that gopacket should behave differently.